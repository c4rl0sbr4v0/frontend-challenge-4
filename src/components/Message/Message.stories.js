import React from 'react';
import Message from './index';

export default {
  title: 'Components/Message',
  component: Message,
};

export const Default = () => <Message key="rd2d" author="bot" text="Peeeeeep peep moc prrrr" />;
