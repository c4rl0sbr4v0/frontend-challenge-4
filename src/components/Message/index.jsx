import React from 'react';
import PropTypes from 'prop-types';
import Avatar from '../Avatar';

/**
 *  Message: Render a Message
 */
const Message = ({ author, text }) => {
  return (
    <article data-author={author} className="media landbot-message">
      <Avatar />
      <div className="media-content landbot-message-content">
        <div className="content">
          <p>{text}</p>
        </div>
      </div>
    </article>
  );
};

export default Message;

Message.propTypes = {
  author: PropTypes.string,
  text: PropTypes.string,
};
Message.defaultProps = {
  author: '',
  text: '',
};
