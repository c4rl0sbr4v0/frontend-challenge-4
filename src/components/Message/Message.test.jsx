import React from 'react';
import { render } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';
import Message from '.';

test('Message Component', async () => {
  const { container, getByText } = render(<Message autho="r2d2" text="peeep" />);
  expect(getByText('peeep')).toBeDefined();
  expect(container.firstChild.classList.contains('landbot-message')).toBe(true);
  expect(container).toMatchSnapshot();
});
