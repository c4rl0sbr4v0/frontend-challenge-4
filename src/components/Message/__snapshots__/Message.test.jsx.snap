// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Message Component 1`] = `
<div>
  <article
    class="media landbot-message"
    data-author=""
  >
    <figure
      class="media-left landbot-message-avatar"
    >
      <p
        class="image is-64x64"
      >
        <img
          alt=""
          class="is-rounded"
          src="r2.jpg"
        />
      </p>
    </figure>
    <div
      class="media-content landbot-message-content"
    >
      <div
        class="content"
      >
        <p>
          peeep
        </p>
      </div>
    </div>
  </article>
</div>
`;
